tests/test_multiaddr.py

```diff
@@ -125,10 +127,19 @@ def test_invalid(addr_str):
         "/ip4/127.0.0.1/p2p/QmcgpsyWgH8Y8ajJz1Cu72KnS5uo2Aa2LpzU7kinSupNKC/tcp/1234",
         "/unix/a/b/c/d/e",
         "/unix/stdio",
+        "/ip4/127.0.0.1/tcp/127/noise",
         "/ip4/1.2.3.4/tcp/80/unix/a/b/c/d/e/f",
         "/ip4/127.0.0.1/p2p/QmcgpsyWgH8Y8ajJz1Cu72KnS5uo2Aa2LpzU7kinSupNKC/tcp/1234/unix/stdio",
         "/dns/example.com",
         "/dns4/موقع.وزارة-الاتصالات.مصر",
+        "/ip4/127.0.0.1/tcp/443/tls/sni/example.com/http/http-path/foo",
+        "/memory/4",
+        "/http-path/tmp%2Fbar",
+        "/http-path/tmp%2Fbar%2Fbaz",
+        "/http-path/foo",
+        "/ip4/127.0.0.1/tcp/9090/http/p2p-webrtc-direct",
+        "/ip4/127.0.0.1/tcp/127/webrtc-direct",
+        "/ip4/127.0.0.1/tcp/127/webrtc",
```

Comment:
You've added certhash addresses to the invalid test. Probably good to add some to the valid test too.